proto/regen/ecocredit/v1alpha1/query.proto

```diff
@@ -177,3 +212,129 @@ message QueryCreditTypesResponse {
+message QuerySellOrderRequest {
+  // sell_order_id is the id of the requested sell order.
+  uint64 sell_order_id = 1;
+}
+
+// QuerySellOrderResponse is the Query/SellOrder response type.
+message QuerySellOrderResponse {
+  // seller_address is the address of the creator of the sell order
+  string seller_address = 1;
+
+  // sell_order contains all information related to a sell order.
+  SellOrder sell_order = 2;
+}
+
+// QuerySellOrdersByDenomRequest is the Query/SellOrdersByDenom request type.
+message QuerySellOrdersByDenomRequest {
+  // denom is the ecocredit denom to query by
+  string denom = 1;
+
+  // pagination defines an optional pagination for the request.
+  cosmos.base.query.v1beta1.PageRequest pagination = 2;
+}
+
+// QuerySellOrdersByDenomResponse is the Query/SellOrdersByDenom response type.
+message QuerySellOrdersByDenomResponse {
+
+  // seller_addresses contain addresses associated with sell orders in the sell_orders array
+  repeated string seller_addresses = 1;
+
+  // sell_orders is a list of sell orders.
+  repeated SellOrder sell_orders = 2;
+
+  // pagination defines an optional pagination for the response.
+  cosmos.base.query.v1beta1.PageResponse pagination = 3;
+}
+
+// QuerySellOrdersByAddrRequest is the Query/SellOrdersByAddress request type.
+message QuerySellOrdersByAddressRequest {
+  // address is the address which created the sell order.
+  string address = 1;
+
+  // pagination defines an optional pagination for the request.
+  cosmos.base.query.v1beta1.PageRequest pagination = 2;
+}
+
+// QuerySellOrdersByAddressResponse is the Query/SellOrdersByAddressResponse response type.
+message QuerySellOrdersByAddressResponse {
+
+  // sell_orders is a list of sell orders.
+  repeated SellOrder sell_orders = 1;
+
+  // pagination defines an optional pagination for the response.
+  cosmos.base.query.v1beta1.PageResponse pagination = 2;
+}
+
+// QueryBuyOrdersRequest is the Query/BuyOrders request type.
+message QueryBuyOrdersRequest {
+  // pagination defines an optional pagination for the request.
+  cosmos.base.query.v1beta1.PageRequest pagination = 1;
+}
+
+// QueryBuyOrdersResponse is the Query/BuyOrders response type.
+message QueryBuyOrdersResponse {
+
+  // buyer_addresses contain addresses associated with buy orders in the buy_orders array
+  repeated string buyer_addresses = 1;
+
+  // buy_orders is a list of buy orders.
+  repeated BuyOrder buy_orders = 2;
+
+  // pagination defines an optional pagination for the response.
+  cosmos.base.query.v1beta1.PageResponse pagination = 3;
+}
+
+// QueryBuyOrderRequest is the Query/BuyOrder request type.
+message QueryBuyOrderRequest {
+  // buy_order_id is the id of the buy order.
+  uint64 buy_order_id = 1;
+}
+
+// QueryBuyOrderResponse is the Query/BuyOrder response type.
+message QueryBuyOrderResponse {
+
+  // buyer_address is the address of the creator of the buy order
+  string buyer_address = 1;
```

Comment:
Same as above. We should include the `buyer` in the `BuyOrder` and consider removing this field.